src/generate_ray.rs

```diff
@@ -0,0 +1,3 @@
+pub fn generate_ray(Vec3D viewpoint, Scene scene) -> Vec<(Vec3D, Color)> {
```

Comment:
For the generate_ray function, we need both direction and origin as parameters. I'll probably make a camera struct in the future.